querymsc.py

```diff
@@ -151,77 +151,56 @@ def query_annual_precip(self, session):
+    def query_design_temp_percentile(self, session, percentile=0.01):
         """A query to get the 1st percentile of a given month across
         the entire operating history of a station in a range of time. 
         Only the year from start and end times are used to create the
         time frame. All frequencies of observations are used, and the
         regular non-corrected air temperature is being used for this 
         calculation.
         -----------------------------------------
+        Args:
+            session (sqlalchemy Session): session constructed using
+                connection string and engine
+            percentile (float): desire percentile in fraction
         Returns: 
             query (sqlalchemy query): sqlalchemy query constructed 
                 using ORM to query temperature percentiles 
         """
 
-        # construct query table
+        # construct query
         query = (
-                session.query(
-                                (func.percentile_cont(0.01)
+                 session.query(
+                                (func.percentile_cont(percentile)
                                      .within_group(Obs.datum.asc())
                                      .label("temp")),
-                                 func.min(Obs.time).label("time_min"),
-                                 func.max(Obs.time).label("time_max"),
-                                 History.lat,
-                                 History.lon,
-                                 History.station_id)
-                            # select from and make join explicit
-                            .select_from([History, Variable])
-                            .filter(
-                                    and_(Obs.time >= self.start_time,
-                                         Obs.time <= self.end_time)
-                                    )
-                            .filter(
-                                    func.extract("month", Obs.time) == self.month
-                                    )
-                            .filter(
-                                    and_(Variable.standard_name == 'air_temperature',
-                                         Variable.id == 1510)
-                                    )
-                            .group_by(History.lat, 
-                                      History.lon, 
-                                      History.station_id)
+                                func.min(Obs.time).label("time_min"),
+                                func.max(Obs.time).label("time_max"),
+                                History.lat.label("lat"),
+                                History.lon.label("lon"),
+                                History.station_id.label("station_id"),
+                                func.count(Obs.datum).label('obs_count')
+                                 )
+                        .join(History)
+                        .join(Variable, Variable.id == Variable.id)
```

Comment:
The explicit join should be `Variable` to `Obs`, thus: `.join(Variable, Obs.variable_id == Variable.id)`.

`.join(Variable, Variable.id == Variable.id)` does nothing; it's a tautology or a no-op, and I suspect it undoes the join you are actually seeking.

Reply:
Strangely, adding any join with the exception of `.join(Variable, Variable.id == Variable.id)`, (probably because it undoes the `.join(Variable)'), returns a single station's data, and a single row of the table. I also get a single row with `.join(Variable, Obs.vars_id == Variable.id)`.

Removing any join with Variable, and joining solely on History gives the full list of stations I'm seeking.

Reply:
Hmmm. This is an example of "programming by friction" (as my CS prof Frank Ruskey once termed it). Meaning that we don't really understand what is going on here, we are just tweaking things until they appear to work.

Could you please add a `print(query)` statement just before you return the query? The string representation of a SQLAlchemy query object is the SQL it will generate, I am pretty sure. And that will tell us some things that aren't guesses.

Reply:
```
SELECT percentile_cont(%(percentile_cont_1)s) 
WITHIN GROUP (ORDER BY crmp.obs_raw.datum ASC) * %(param_1)s AS temp, min(crmp.obs_raw.obs_time) AS time_min, max(crmp.obs_raw.obs_time) AS time_max, crmp.meta_history.lat AS lat, crmp.meta_history.lon AS lon, crmp.meta_history.station_id AS station_id, count(crmp.obs_raw.datum) AS obs_count 
FROM crmp.meta_vars, crmp.obs_raw JOIN crmp.meta_history ON crmp.obs_raw.history_id = crmp.meta_history.history_id 
WHERE crmp.obs_raw.obs_time >= %(obs_time_1)s AND crmp.obs_raw.obs_time < %(obs_time_2)s AND EXTRACT(month FROM crmp.obs_raw.obs_time) = %(param_2)s AND crmp.meta_vars.vars_id = %(vars_id_1)s AND crmp.obs_raw.datum != %(datum_1)s 
GROUP BY crmp.meta_history.lat, crmp.meta_history.lon, crmp.meta_history.station_id
```

Reply:
@rod-glover I've updated this previous comment a lot. I was missing some crucial information. Tagging you just to make sure you get the notification for this updated version.

Reply:
Hmmm, it seems that both of the above queries are returning all stations. I believe that adding the `select_from(Obs).join(Variable, Obs.vars_id == Variable.id)` which was a recommendation from another conversation, seems to have solved this problem.

Reply:
Ahh, I think I see what is going on. I've changed my filtering methods from `Variable.id` to a combination of `Variable.standard_name` and `Variable.cell_method`. I'm going to resolve this and request a review on another PR.